src/app/ApplicationTemplate.Presentation/ViewModels/ViewModel.cs

```diff
@@ -18,20 +18,29 @@ namespace ApplicationTemplate.Presentation;
 public class ViewModel : ViewModelBase, INavigableViewModel
 {
 	// Add properties or commands you want to have on all your ViewModels
+	private readonly ISectionsNavigator _sectionsNavigator;
 
 	public ViewModel()
 	{
 		(this as IInjectable)?.Inject((t, n) => this.GetService(t));
+		_sectionsNavigator = this.GetService<ISectionsNavigator>();
+	}
+
+	public bool IsModalActive
```

Comment:
It doesn't feel like the good solution. It looks like any app that uses modals will have to manipulate flags on the pages behind the modals. Modals can be opened from anywhere so there are cases where you don't know in advance that page is behind.